static/js/show-worklist.js

```diff
@@ -8,4 +8,25 @@ function search_by_username() {
     $('#search_by_worklist_name').addClass('hidden');
     $('#search_by_username').removeClass('hidden');
     $('#dropdownMenuButton').html('By Username');
-}
\ No newline at end of file
+}
+
+(window).on('load', function worker() {
+     if($('#search_by_worklist_name').is(":visible") == true) {
+                search_term = document.getElementById("search_by_worklist_name_form").search_term.value
+                search_type = document.getElementById("search_by_worklist_name_form").search_type.value
+            } else {
+                search_term = document.getElementById("search_by_username_form").search_term.value
+                search_type = document.getElementById("search_by_username_form").search_type.value
+            }
+
+            $('#worklist-table').html('');
+            $('#worklist-table').addClass('loader');
+            $('#worklist-table').html('').load(
```

Comment:
We discussed this a little but let me write it down as well since anything that's not written gets lost. Reloading the entire table is enough for now, but probably still bad UX in the long run.

What we probably want in the longer term is to separate the live updates in two kinds, from a user's perspective: "light" ones, which affect only a column of the table, and "heavy" ones, which add/remove rows.

So the "light" ones are are updates to progress or who's claimed the worklist. These update the corresponding table cells automatically. The "heavy" would be things like adding new articles manually, for which we should possibly show a note or button in the UI that warns the user that there are updates and allows them to refresh, without actually refreshing all of the data suddenly.

The "heavy" updates functionality is not super necessary in the foreseeable future though, so effectively updating the full table is fine right now.

Reply:
Thanks for putting it down :).